pyanaconda/dbus/namespace.py

```diff
@@ -0,0 +1,176 @@
+        """Describe a DBus object.
+
+        :param name: a namespace sequence
+        :param namespace: a namespace prefix
+        :param interface_version: a version of the DBus interface
+        :param object_version: a version of the DBus object
+        """
+        super().__init__(*name, namespace=namespace, interface_version=interface_version)
+        self._object_version = object_version
+
+    @property
+    def object_path(self):
+        """Full path of the DBus object."""
+        return self.path_space + self.version_to_string(self._object_version)
+
+
+class DBusServiceIdentifier(DBusObjectIdentifier):
+    """Identifier of a DBus service."""
+
+    def __init__(self, *name, namespace=None, interface_version=None, object_version=None,
+                 service_version=None, message_bus=DBus):
+        """Describe a DBus service.
+
+        :param name: a namespace sequence
+        :param namespace: a namespace prefix
+        :param interface_version: a version of the DBus interface
+        :param object_version: a version of the DBus object
+        :param service_version: a version of the DBus service
+        :param message_bus: a message bus
+        """
+        super().__init__(*name,
+                         namespace=namespace,
+                         interface_version=interface_version,
+                         object_version=object_version)
+
+        self._service_version = service_version
+        self._message_bus = message_bus
+
+    @property
+    def service_name(self):
+        """Full name of a DBus service."""
+        return self.name_space + self.version_to_string(self._service_version)
+
+    def get_proxy(self, object_path=None):
+        """Returns a proxy of the DBus object.
+
+        :param object_path: a DBus path an object
+        :return: a proxy object
+        """
+        if object_path is None:
+            object_path = self.object_path
+
+        return self._message_bus.get_proxy(self.service_name, object_path)
+
+    def get_observer(self, object_path=None):
+        """Returns an observer of the DBus object.
+
+        :param object_path: a DBus path of an object
+        :return: an observer object
+        """
+        if object_path is None:
+            object_path = self.object_path
+
+        return self._message_bus.get_observer(self.service_name, object_path)
+
+    def get_cached_observer(self, object_path=None, interface_names=None):
+        """Returns a cached observer of the DBus object.
+
+        :param object_path: a DBus path of an object
+        :param interface_names: a list of interface names
+        :return: an observer object
+        """
+        if object_path is None:
+            object_path = self.object_path
+
+            if interface_names is None:
+                interface_names = [self.interface_name]
+
+        return self._message_bus.get_cached_observer(
+            self.service_name, object_path, interface_names
```

Comment:
The service identifier has these methods, so it can be used as a shortcut. Then we can call 

TIMEZONE.get_proxy() 

instead of 

DBus.get_proxy(TIMEZONE.service_name, TIMEZONE.object_path)

If you prefer the second option, I can remove this functionality from the service identifier.

Reply:
Yeah, I think that shortcut could be really valuable. You can leave it there.